### PR TITLE
Stash Pull Request Support

### PR DESCRIPTION
**Pull request by GordonDiggs**
This will allow us to update PRs on Stash instances.

Documentation for the Stash build status API:
https://developer.atlassian.com/stash/docs/latest/how-tos/updating-build-status-for-commits.html

@codeclimate/review

### CODE REVIEW

File: lib/cc/services/stash_pull_requests.rb [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+  def commit_sha
+    @payload.fetch("commit_sha")
+  end
+
+  def description
+    return @payload["message"] if @payload["message"]
+
+    case @payload["state"]
+    when "pending"
+      "Code Climate is analyzing this code."
```
Comment (fhwang) [inline lib/cc/services/stash_pull_requests.rb]: These message descriptions seem like prime candidates for DRYing across the two kinds of PRs.

Comment (GordonDiggs) [inline lib/cc/services/stash_pull_requests.rb]: Good call - updated

File: lib/cc/services/github_pull_requests.rb [hunk 2]
```diff
@@ -87,18 +81,18 @@ def update_status_failure
...
     update_status(
       "error",
-      @payload["message"] || DEFAULT_ERROR
+      @payload["message"] || presenter.error_message
     )
   end
 
   def update_status_pending
-    update_status("pending", "Code Climate is analyzing this code.")
+    update_status("pending", presenter.pending_message)
```
Comment (pbrisbin) [inline lib/cc/services/github_pull_requests.rb]: There's no harm in adding the `@payload["message"] ||` part here now right? I'm going to need it soon.

Comment (GordonDiggs) [inline lib/cc/services/github_pull_requests.rb]: Correct. I can add that!

**Comment by GordonDiggs** [comment]
@codeclimate/review ready for another look!

**Comment by pbrisbin** [comment]
LGTM

**Pull request merged**